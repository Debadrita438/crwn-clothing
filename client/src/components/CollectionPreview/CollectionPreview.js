import { withRouter } from 'react-router-dom';
import CollectionItem from '../CollectionItem/CollectionItem';

import { CollectionPreviewContainer, PreviewContainer, TitleStyles } from './CollectionPreview.styles';

export const CollectionPreview = ({ title, items, history, match, routeName }) => (
    <CollectionPreviewContainer>
        <TitleStyles onClick={() => history.push(`${match.path}/${ routeName }`)}>
            { title.toUpperCase() }
        </TitleStyles>
        <PreviewContainer>
            {
                items
                    .filter((item, i) => i < 4)
                    .map( item => (
                        <CollectionItem key={ item.id } item={ item } />
                ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
);

 
export default withRouter(CollectionPreview);