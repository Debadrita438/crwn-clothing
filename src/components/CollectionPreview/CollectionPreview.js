import CollectionItem from '../CollectionItem/CollectionItem';

import { CollectionPreviewContainer, PreviewContainer, TitleStyles } from './CollectionPreview.styles';

const CollectionPreview = ({ title, items }) => (
    <CollectionPreviewContainer>
        <TitleStyles>{ title.toUpperCase() }</TitleStyles>
        <PreviewContainer>
            {
                items
                    .filter((item, i) => i < 4)
                    .map( item => (
                        <CollectionItem key={ item.id } item={ item } />
                ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
);

 
export default CollectionPreview;