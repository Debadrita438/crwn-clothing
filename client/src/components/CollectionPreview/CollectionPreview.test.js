import { shallow } from 'enzyme';

import { CollectionPreview } from './CollectionPreview';

describe('CollectionPreview Component', () => {
    let wrapper;
    let mockHistory;
    let mockTitle = 'Hats';
    let mockMatch;
    let mockRouteName = 'hats';

    beforeEach(() => {
        mockHistory = {
            push: jest.fn()
        };

        mockMatch = { 
            path: '/shop'
        };

        const mockProps = {
            title: mockTitle,
            items: [],
            history: mockHistory,
            match: mockMatch,
            routeName: mockRouteName
        };

        wrapper = shallow(<CollectionPreview { ...mockProps } />);
    });

    it('should render the CollectionPreview Component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the collection preview page of the title', () => {
        wrapper.find('TitleStyles').simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(`${ mockMatch.path }/${ mockRouteName }`);
    });
});