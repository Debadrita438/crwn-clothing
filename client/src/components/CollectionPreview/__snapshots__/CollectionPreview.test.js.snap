// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview Component should render the CollectionPreview Component 1`] = `ShallowWrapper {}`;
